### PR TITLE
Show resource count when clicking on cards in LogPanel

### PR DESCRIPTION
**Pull request by nwai90**
Display resource count for cards clicked from LogPanel, if any

Screenshot 2021-09-10 at 11 39 36 AM

### CODE REVIEW

**Review by alrusdi — APPROVED** [review]

File: src/components/LogPanel.vue [hunk 1]
```diff
@@ -334,6 +334,16 @@ export default Vue.extend({
         current: PartyName.GREENS,
       };
     },
+    getResourcesOnCard: function(cardName: CardName) {
```
Comment (Argeento) [inline src/components/LogPanel.vue:337]: please use shorter syntax like in every other methods
```js
 getResourcesOnCard(cardName: CardName) {
 ```

File: src/components/LogPanel.vue [hunk 2]
```diff
@@ -334,6 +334,16 @@ export default Vue.extend({
...
+      let resourceCount = undefined;
+
+      for (const player of this.players) {
+        const foundCard = player.playedCards.find((card) => card.name === cardName);
+        if (foundCard !== undefined) resourceCount = foundCard.resources;
```
Comment (Argeento) [inline src/components/LogPanel.vue:342]:
```js
if (foundCard) return foundCard.resoruce;
```
You don't have to create `resourceCount` variable. Moreover `for` loop can stop if card is found. (If you like this level of clarity with additional variable, just add `break`)

File: src/components/LogPanel.vue [hunk 3]
```diff
@@ -334,6 +334,16 @@ export default Vue.extend({
         current: PartyName.GREENS,
       };
     },
+    getResourcesOnCard: function(cardName: CardName) {
+      let resourceCount = undefined;
```
Comment (Argeento) [inline src/components/LogPanel.vue:338]: It's
```js
let resourceCount: number | undefined;
```
not 
```
let resourceCount = undefined;
```
Now you are creating variable with `any` type. Btw `undefined` is assign to variable by default. You don't have to explicitly define it.

**Review by Argeento — CHANGES_REQUESTED** [review]

File: src/components/LogPanel.vue (hunk 2 above)
Comment (kberg) [reply src/components/LogPanel.vue]: This is a good tip. @nwai90 mind including this?

**Review by kberg — APPROVED** [review]

File: src/components/LogPanel.vue (hunk 2 above)
Comment (nwai90) [reply src/components/LogPanel.vue]: Agree. I updated the code for early `return`

**Pull request merged**